lib/plugins/cannulaage.js

```diff
@@ -4,52 +4,96 @@ var _ = require('lodash');
-  cage.pluginType = 'pill-minor';
-
-  cage.updateVisualisation = function updateVisualisation (sbx) {
-    var age = 0;
-    var found = false;
-    var treatmentDate = null;
-    var message = '';
-
-    _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
-      if (treatment.eventType === 'Site Change') {
-        treatmentDate = new Date(treatment.created_at);
-        var hours = Math.round(Math.abs(sbx.time - treatmentDate) / 36e5);
-
-        if (!found) {
-          found = true;
-          age = hours;
-        } else {
-          if (hours < age) {
-            age = hours;
-            if (treatment.notes) {
-              message = treatment.notes;
-            }
-          }
-        }
+  var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+  if (!cannulaInfo.checkForAlert) {
+    return;
+  }
+
+  // CAGE_WARN=48 CAGE_URGENT=70
+
+  if (cannulaInfo.age === sbx.extendedSettings.urgent || cannulaInfo.age === sbx.extendedSettings.warn )
+  {
+    var title = 'Cannula age ' + cannulaInfo.age +' hours';
+    sbx.notifications.requestNotify({
+      level: 2
+      , title: title
+      , message: title + ', time to change?'
+      , pushoverSound: 'persistent'
+      , plugin: cage
+      , debug: {
+        cannulaInfo: cannulaInfo
       }
     });
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+  }
+}
 
-    sbx.pluginBase.updatePillText(cage, {
-      value: age + 'h'
-      , label: 'CAGE'
-      , info: info
-    });
+function frac(f) {
+  return f % 1;
+}
 
-  };
+cage.findLatestTimeChange = function findLatestTimeChange(sbx) {
 
-  return cage();
+  var returnValue = {'message':'', 'found': false, 'age': 0, 'treatmentDate': null, 'checkForAlert': false};
+
+  _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
+    if (treatment.eventType === 'Site Change') {
+      returnValue.treatmentDate = new Date(treatment.created_at);
+      var hours = Math.round(Math.abs(sbx.time - returnValue.treatmentDate) / 36e5);
+
+      //allow for around 10 minute period after a full hour during which we'll alert the user
+      returnValue.checkForAlert = frac(Math.abs(sbx.time - returnValue.treatmentDate) / 36e5) < 0.15;
+
+      if (!returnValue.found) {
+        returnValue.found = true;
+        returnValue.age = hours;
+      } else {
+        if (hours < returnValue.age) {
+          returnValue.age = hours;
+          if (treatment.notes) {
+            returnValue.message = treatment.notes;
+          }
+        }
+      }
+    }
+  });
+
+  return returnValue;
 }
 
+cage.updateVisualisation = function updateVisualisation (sbx) {
+
+  var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+  var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+  if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511228/issues/source?bid=2452379&fileBranchId=2457543#l90)